projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -85,11 +96,42 @@ export class CartesianLegend {
   }
 
   private drawLegendEntries(container: ContainerElement): void {
-    select(container)
+    const containerSelection = select(container);
+    if (!this.isGrouped) {
+      containerSelection
+        .append('div')
+        .classed('legend-entries', true)
+        .selectAll('.legend-entry')
+        .data(this.series.filter(series => !series.hide))
+        .enter()
+        .each((_, index, elements) => this.drawLegendEntry(elements[index]));
+    } else {
+      containerSelection
+        .selectAll('.legend-entries')
+        .data(this.groupedSeries)
+        .enter()
+        .append('div')
+        .attr('class', seriesGroup => `legend-entries group-${seriesGroup[0].assignedGroup!.id}`)
+        .each((seriesGroup, index, elements) => this.drawLegendEntriesTitleAndValues(seriesGroup, elements[index]));
+    }
+  }
+
+  private drawLegendEntriesTitleAndValues(seriesGroup: Series<{}>[], element: HTMLDivElement): void {
```

Comment:
may be do `Series<TData>` or `Series<unknown>`

Reply:
Done